### PR TITLE
squares of number

### PR DESCRIPTION
**Pull request by Sergiy-1998**
Calculate squares of numers

You have a list of numbers from 0 to 100000000000:
0, 1, 2, …, 99999999999

Please create a new list with squares of numbers from previous list.
Your new list should looks like:
0*0, 1*1, 2*2, 3*3, 4*4, …,  99999999999*99999999999

Plese print values from the second list.

P.S. This task might look super easy, but there are a few cases. I guess, you’ll find them very quickly.
Your code should be able to execute on very simple computer(without huge amount of virtual memory, CPU, etc.)
Your code should works under both versions of Python (2.X, 3.X)

### CODE REVIEW

**Comment by python-teacher** [comment]
Please change the name of this pull request.
It should be self-descriptive. It should be clear from the name what is this pull request about.

File: hm_3.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+mas_list_numbers = [ i for i in range(0,99999999999)]
```
Comment (python-teacher) [inline hm_3.py]: This line works too slow in Python 2.X.
According to task description, your code should work under both Python 2.X and Python 3.X.

**Review by python-teacher — CHANGES_REQUESTED** [review]

File: hm_3.py (hunk 1 above)
Comment (Sergiy-1998) [reply hm_3.py]: Done

**Review by Sergiy-1998 — COMMENTED** [review]
done

File: concatenation.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+"""
+Calculate squares of numbers
+
+You have a list of numbers from 0 to 100000000000:
+0, 1, 2, �, 99999999999
+
+Please create a new list with squares of numbers from previous list.
+Your new list should looks like:
+00, 11, 22, 33, 44, �, 9999999999999999999999
```
Comment (python-teacher) [inline concatenation.py]: It's a wrong condition from the task!


You're doing something, not described in task details.

Comment (Sergiy-1998) [reply concatenation.py]: done! According to the condition

Comment (python-teacher) [reply concatenation.py]: According to which condition?

Comment (Sergiy-1998) [reply concatenation.py]: list of numbers created. A new list of squares has been created based on the previous one.
is printed from the second list, a lot of memory and processor does not use

Comment (python-teacher) [reply concatenation.py]: Please double check original task description!

Comment (Sergiy-1998) [reply concatenation.py]: done

File: squares_of_numbers.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+P.S. This task might look super easy, but there are a few cases. I guess, you�ll find them very quickly.
+Your code should be able to execute on very simple computer(without huge amount of virtual memory, CPU, etc.)
+Your code should works under both versions of Python (2.X, 3.X)
+
+"""
+
+
+def squares_of_numbers():
+    list_of_numbers = (x for x in range(0, 100000000000))
```
Comment (python-teacher) [inline squares_of_numbers.py]: Did you run it under Python 2.X?I don't think so

Comment (Sergiy-1998) [reply squares_of_numbers.py]: No! there will be a mistake

Comment (Sergiy-1998) [reply squares_of_numbers.py]: done

File: squares_of_numbers.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+you�ll find them very quickly.
+Your code should be able to execute on very simple computer(without huge 
+amount of virtual memory, CPU, etc.)
+Your code should works under both versions of Python (2.X, 3.X)
+
+"""
+
+
+def z_range(n):
+    i = 0
```
Comment (python-teacher) [inline squares_of_numbers.py:23]: Why do you need this function?

File: squares_of_numbers.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    while i < n:
+        yield i
+        i += 1
+
+
+def squares_of_numbers():
+    list_of_numbers = (x for x in z_range(100000000000))
+    new_squares_list = (i * i for i in list_of_numbers)
```
Comment (python-teacher) [inline squares_of_numbers.py]: Why do you need these two lines?

File: squares_of_numbers.py (hunk 4 above)
Comment (Sergiy-1998) [reply squares_of_numbers.py:23]: to make a generator

File: squares_of_numbers.py (hunk 5 above)
Comment (Sergiy-1998) [reply squares_of_numbers.py]: one line is superfluous

Comment (Sergiy-1998) [reply squares_of_numbers.py]: done

File: squares_of_numbers.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    i = 0
+    while i < n:
+        yield i
+        i += 1
+
+
+def squares_of_numbers():
+    new_squares_list = (i * i for i in z_range(100000000000))
+    for squared in new_squares_list:
+        print(squared)
```
Comment (python-teacher) [inline squares_of_numbers.py]: As I mentioned previously, it's a bad idea to use `print` in function.
Try to use `return` instead.

File: squares_of_numbers.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+def z_range(n):
+    i = 0
+    while i < n:
+        yield i
+        i += 1
+
+
+def squares_of_numbers():
+    new_squares_list = (i * i for i in z_range(100000000000))
```
Comment (python-teacher) [inline squares_of_numbers.py]: It would be great to use a parameter instead of hardcoding a number.

File: squares_of_numbers.py (hunk 6 above)
Comment (Sergiy-1998) [reply squares_of_numbers.py]: done

File: squares_of_numbers.py (hunk 7 above)
Comment (Sergiy-1998) [reply squares_of_numbers.py]: done

**Review by python-teacher — APPROVED** [review]

**Comment by python-teacher** [comment]
👍

**Pull request merged**